src/libraries/System.Linq/src/System/Linq/Chunk.SpeedOpt.cs

```diff
@@ -0,0 +1,39 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace System.Linq
+{
+    public static partial class Enumerable
+    {
+        private static IEnumerable<TSource[]> ChunkIteratorOptimized<TSource>(IEnumerable<TSource> source, int maxSize, int totalCount)
```

Comment:
This function should just be in Chunk.cs rather than creating this separate SpeedOpt file.